doc/source/io.rst

```diff
@@ -440,6 +440,45 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+.. note::
+
+   Reading in data with mixed dtypes and relying on ``pandas``
+   to infer them is not recommended. In doing so, the parsing engine will
+   loop over all the dtypes, trying to convert them to an actual
+   type; if something breaks during that process, the engine will go to the
+   next ``dtype`` and the data is left modified in place. For example,
+
+   .. ipython:: python
+
```

Comment:
Add here a `:okwarning:`

like

```
.. ipython:: python
    :okwarning:

    df = ...
```

This is to let sphinx know that it does not have to complain about this code raising a warning (as the readd_csv will do)